File: src/lib/fly/versions.ts
/**
 * FLY バージョン定数。
 * Edge / Fly backend の ANALYZER_VERSION と揃えてキャッシュを制御する。
 */

import { REMOTE_ANALYZER_VERSION } from "../songAnalyzeClient";
import type { FlyVersionBundle } from "./types";

/** 複数ソース統合ロジック */
export const FLY_FUSION_VERSION = "fly-fusion-v0.1.0";

/** ダンス知能（Opportunity / Impact / Urgency） */
export const FLY_DANCE_MODEL_VERSION = "fly-dance-v0.1.0";

/** Formation Tier1 接続レイヤ（既存 engine を壊さない） */
export const FLY_FORMATION_MODEL_VERSION = "choreocore-tier1-v5";

/** Structure v2 / AIO の論理バージョン（ドキュメント用） */
export const FLY_STRUCTURE_V2_VERSION = "structure-v2.0.1";
export const FLY_AIO_VERSION = "all-in-one-v1.0.0";

export function currentFlyVersions(
  analyzerVersion: string = REMOTE_ANALYZER_VERSION
): FlyVersionBundle {
  return {
    analyzerVersion,
    fusionVersion: FLY_FUSION_VERSION,
    danceModelVersion: FLY_DANCE_MODEL_VERSION,
    formationModelVersion: FLY_FORMATION_MODEL_VERSION,
  };
}

File: backend/analyzer/fly_music_intelligence/versions.py
"""
FLY Music Intelligence — version bundle (Dance Music Intelligence Engine).

既存 services.* の定数と揃え、キャッシュ無効化は analyzer_version で行う。
"""

from __future__ import annotations

# 既存 v1 キャッシュキー（Edge / client と同期）
ANALYZER_VERSION = "algo-v1.5.0"
STRUCTURE_V2_VERSION = "structure-v2.0.1"
AIO_VERSION = "all-in-one-v1.0.0"

FUSION_VERSION = "fly-fusion-v0.1.0"
DANCE_MODEL_VERSION = "fly-dance-v0.1.0"
FORMATION_MODEL_VERSION = "choreocore-tier1-v5"


def version_bundle() -> dict[str, str]:
    return {
        "analyzer_version": ANALYZER_VERSION,
        "structure_v2_version": STRUCTURE_V2_VERSION,
        "aio_version": AIO_VERSION,
        "fusion_version": FUSION_VERSION,
        "dance_model_version": DANCE_MODEL_VERSION,
        "formation_model_version": FORMATION_MODEL_VERSION,
    }
